dev-libs/boost/boost166-1.66.0.recipe

```diff
@@ -0,0 +1,180 @@
+	devel:libboost_date_time$secondaryArchSuffix = $portVersion
+	devel:libboost_exception$secondaryArchSuffix = $portVersion
+	devel:libboost_filesystem$secondaryArchSuffix = $portVersion
+	devel:libboost_graph$secondaryArchSuffix = $portVersion
+	devel:libboost_iostreams$secondaryArchSuffix = $portVersion
+	devel:libboost_locale$secondaryArchSuffix = $portVersion
+	devel:libboost_log$secondaryArchSuffix = $portVersion
+	devel:libboost_log_setup$secondaryArchSuffix = $portVersion
+	devel:libboost_math_c99$secondaryArchSuffix = $portVersion
+	devel:libboost_math_c99f$secondaryArchSuffix = $portVersion
+	devel:libboost_math_tr1$secondaryArchSuffix = $portVersion
+	devel:libboost_math_tr1f$secondaryArchSuffix = $portVersion
+	devel:libboost_prg_exec_monitor$secondaryArchSuffix = $portVersion
+	devel:libboost_program_options$secondaryArchSuffix = $portVersion
+	devel:libboost_random$secondaryArchSuffix = $portVersion
+	devel:libboost_regex$secondaryArchSuffix = $portVersion
+	devel:libboost_serialization$secondaryArchSuffix = $portVersion
+	devel:libboost_signals$secondaryArchSuffix = $portVersion
+	devel:libboost_stacktrace_basic$secondaryArchSuffix = $portVersion
+	devel:libboost_stacktrace_noop$secondaryArchSuffix = $portVersion
+	devel:libboost_system$secondaryArchSuffix = $portVersion
+	devel:libboost_test_exec_monitor$secondaryArchSuffix = $portVersion
+	devel:libboost_thread$secondaryArchSuffix = $portVersion
+	devel:libboost_timer$secondaryArchSuffix = $portVersion
+	devel:libboost_type_erasure$secondaryArchSuffix = $portVersion
+	devel:libboost_unit_test_framework$secondaryArchSuffix = $portVersion
+	devel:libboost_wave$secondaryArchSuffix = $portVersion
+	devel:libboost_wserialization$secondaryArchSuffix = $portVersion
+	"
+
+REQUIRES_devel="
+	haiku$secondaryArchSuffix >= r1~alpha4_pm_hrev51470
+	boost165$secondaryArchSuffix == $portVersion
+	"
+
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	devel:libbz2$secondaryArchSuffix
+	devel:libz$secondaryArchSuffix
+	devel:libicuuc$secondaryArchSuffix
+	devel:libicui18n$secondaryArchSuffix
+	devel:libicudata$secondaryArchSuffix
+	"
+
+BUILD_PREREQUIRES="
+	cmd:gcc$secondaryArchSuffix
+	cmd:ld$secondaryArchSuffix
+	cmd:sed
+	cmd:which
+	"
+
+BUILD()
+{
+	./bootstrap.sh \
+		--prefix=$prefix \
+		--exec-prefix=$binDir \
+		--libdir=$libDir \
+		--includedir=$includeDir
+
+	./b2 -q $jobArgs \
+		--without-mpi \
+		--enable-threads=posix \
+		--enable-thread-local-alloc \
+		--enable-parallel-mark \
+		inlining=on \
+		threading=multi \
+		variant=release \
+		link=static,shared \
+		runtime-link=shared \
+		--without-python
+}
+
+INSTALL()
+{
+	./b2 -q $jobArgs \
+		--without-mpi \
+		--enable-threads=posix \
+		--enable-thread-local-alloc \
+		--enable-parallel-mark \
+		inlining=on \
+		threading=multi \
+		variant=release \
+		link=static,shared \
+		runtime-link=shared \
+		--without-python \
+		install
+
+	prepareInstalledDevelLibs libboost_atomic libboost_chrono \
```

Comment:
Given the huge number of libs, you might wish to replace lines 159 to 172 by these 2 lines:
```
	prepareInstalledDevelLibs `echo "$PROVIDES" | sed -n \
		-e "s/lib:\(.*\)$secondaryArchSuffix =.*/\1/p"`
```

Reply:
Nope, thats incomplete. Tested with PROVIDES_Devel, but that also not the complete list.

One have to merge PROVIDES and PROVIDES_devel, eliminate the duplicates and sort it in abc. Then it should work.

Reply:
Good catch. I had not noticed that the `PROVIDES` and `PROVIDES_devel` did not match exactly.

It seems there are 31 `lib:lib*` in `PROVIDES` and 33 `devel:lib*` in `PROVIDES_devel`. Well, since the call to `prepareInstalledDevelLibs` mentions these 33 libs, this variant should work:
```
	prepareInstalledDevelLibs `echo "$PROVIDES_devel" | sed -n \
		-e "s/devel:\(.*\)$secondaryArchSuffix =.*/\1/p"`
```

FWIW, the 2 "extra" libs that are only in `PROVIDES_devel` are `libboost_exception` and `libboost_test_exec_monitor`.

Reply:
Thanks it worked.